From 3ec983218703b69424b47037134ccd130490522d Mon Sep 17 00:00:00 2001
From: jinghua <jinghua@marvell.com>
Date: Wed, 16 Dec 2015 15:44:01 +0800
Subject: [PATCH 0604/1240] fix: i2c: a3700: add page write delay to i2c mm
 command

    In i2c mm command, after each page write, need to delay 10 msec
    before perform the next read, according to EEPROM ST M24C64 spec,
    otherwise the next read operation fails.

    Add define CONFIG_SYS_EEPROM_PAGE_WRITE_DELAY_MS in file armadalp.h,
    which would be used in common/cmd_i2c.c, routine mod_i2c_mem.

Change-Id: Ia569fc7570240a49eba341250e403029e81b13f2
Signed-off-by: jinghua <jinghua@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/25994
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 include/configs/armadalp.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/configs/armadalp.h b/include/configs/armadalp.h
index 7b719f8..264a55b 100644
--- a/include/configs/armadalp.h
+++ b/include/configs/armadalp.h
@@ -97,6 +97,12 @@ extern char __end_of_spl[];
 
 #endif /* CONFIG_MVEBU_SPI */
 
+/* The EEPROM ST M24C64 has 32 byte page write mode and takes up to 10 msec.
+   In u-boot I2C mm command, after each page write, need to deley 10 msec to
+   perform the next read.
+ */
+#define CONFIG_SYS_EEPROM_PAGE_WRITE_DELAY_MS	10
+
 /*
  * Include the generic MVEBU configuration file.
  * The generic file is affected by the defines above.
-- 
1.9.1

